helix-vcs/src/lib.rs

```diff
@@ -38,6 +46,13 @@ impl DiffProviderRegistry {
             .iter()
             .find_map(|provider| provider.get_diff_base(file))
     }
+
+    pub fn get_changed_files(&self, cwd: &Path) -> Result<Vec<FileChange>> {
+        self.providers
+            .iter()
+            .find_map(|provider| provider.get_changed_files(cwd).ok())
+            .ok_or_else(|| anyhow::anyhow!("no diff provider returns success"))
```

Comment:
I find this better than using a negation, it avoids having 'success' as the last word

```suggestion
            .ok_or_else(|| anyhow::anyhow!("all diff providers failed to list changed files"))
```